spec.md

```diff
@@ -150,33 +153,120 @@ where $Q$ is the largest integer such that
 (Note:
 When $|R(X)| > 0$,
 $L(X)$ is non-negative,
-because $P(X)$ is defined in terms of a hash with $T$ trailing zeroes.
-But when $|R(X)| = 0$,
+because $P(X)$ is defined in terms of a hash with $T$ trailing zeroes
+(except when the split is triggered by hitting $S_\text{max}$).
+But when $|R(X)| = 0$
+(or in the $S_\text{max}$ case),
 that hash may have fewer than $T$ trailing zeroes,
 and so $L(X)$ may be negative.
 This makes no difference to the algorithm below, however.)
 
 A “node” in a hashsplit tree
-is a pair $(D, C)$
+is a pair $(D, S)$
 where $D$ is the node’s “depth”
-and $C$ is a sequence of children.
+and $S$ is a sequence of children.
 The children of a node at depth 0 are chunks
 (i.e., subsequences of the input).
 The children of a node at depth $D > 0$ are nodes at depth $D - 1$.
 
-The function $\operatorname{Children}(N)$ on a node $N = (D, C)$ produces $C$
+The function $\operatorname{Children}(N)$
+on a node $N = (D, S)$
+produces $S$
 (the sequence of children).
 
+The function $\operatorname{Rightmost}(N)$
+on a node $N = (D, \langle S_0, \dots, S_b \rangle)$
+produces the “rightmost leaf chunk”
+defined recursively as follows:
+
+- If $D = 0$,
+  $\operatorname{Rightmost}(N) = K_{C,b}$
+- Otherwise, $\operatorname{Rightmost}(N) = \operatorname{Rightmost}(S_b)$
+
+
 ## Algorithm
 
+This section contains two descriptions of hashsplit trees:
+an algebraic description for formal reasoning,
+and a procedural description for practical construction.
+
+### Algebraic description
+
+Let function $F_C(X, D, i)$ on a sequence $X$, a depth $D$, and an index $i$ produce the $i^\text{th}$ node at level $D$.
+
+Then the root of the tree is $F_C(X, D, 0)$
```

Comment:
This could alternatively be defined as the lowest node F_C(X, D, 0) such that Rightmost(F_C(X, D, 0)) = the last chunk in SPLIT_C(X). Which is better?

Reply:
I think the current phrasing is fine.